File: C/media_pares.c
#include <stdio.h>
#include <stdlib.h>

int main() {

    int N, soma = 0, contpares = 0;
    double mediapares;

    printf("Quantos elementos vai ter o vetor? ");
    scanf("%d", &N);

    int vet[N];

    for(int i = 0; i < N; i++){
        printf("Digite um numero: ");
        scanf("%d", &vet[i]);
    }

    for(int i = 0; i < N; i++) {
        if(vet[i] % 2 == 0) {
            soma = soma + vet[i];
            contpares = contpares + 1;
        }
    }

    if (contpares == 0){
        printf("NENHUM NUMERO PAR\n");
    }else {
        mediapares = (double)soma / contpares;
        printf("MEDIA DOS PARES: %.1lf", mediapares);
    }

    return 0;
}

File: java/media_pares.java
import java.util.Locale;
import java.util.Scanner;

public class media_pares {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int n, somapar,npares;
		double media;
		
		System.out.print("Quantos elementos vai ter o vetor? ");
		n = sc.nextInt();
		
		int[] vet = new int[n]; //Representação do vetor
		
		for (int i = 0; i < n; i++) {
			System.out.print("Digite um numero: ");
			vet[i] = sc.nextInt();
		}
		
		
		somapar = 0;
		npares = 0;
		for (int i = 0; i < n; i++) {
			if (vet[i] % 2 == 0) {
				somapar = somapar + vet[i];
				npares = npares + 1;
			}
		}
		
		if (npares == 0) {
			System.out.println("NENHUMA NUMERO PAR");			
		} else {
			media = (double) somapar / npares;
			System.out.print("MEDIA DOS PARES = " + String.format("%.1f", media));
		}
		
		sc.close();
	}

}
